Userland/Libraries/LibJS/Runtime/RegExpConstructor.cpp

```diff
@@ -27,6 +38,94 @@ void RegExpConstructor::initialize(Realm& realm)
     define_native_accessor(realm, *vm.well_known_symbol_species(), symbol_species_getter, {}, Attribute::Configurable);
 
     define_direct_property(vm.names.length, Value(2), Attribute::Configurable);
+
+    // Additional properties of the RegExp constructor, https://github.com/tc39/proposal-regexp-legacy-features#additional-properties-of-the-regexp-constructor
+    define_native_accessor(realm, "input", input, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$_", input, {}, Attribute::Configurable);
+    define_native_accessor(realm, "lastMatch", last_match, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$&", last_match, {}, Attribute::Configurable);
+    define_native_accessor(realm, "lastParen", last_paren, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$+", last_paren, {}, Attribute::Configurable);
+    define_native_accessor(realm, "leftContext", left_context, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$`", left_context, {}, Attribute::Configurable);
+    define_native_accessor(realm, "rightContext", right_context, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$'", right_context, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$1", $1, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$2", $2, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$3", $3, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$4", $4, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$5", $5, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$6", $6, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$7", $7, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$8", $8, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$9", $9, {}, Attribute::Configurable);
+}
+
+// UpdateLegacyRegExpStaticProperties ( C, S, startIndex, endIndex, capturedValues )
+// https://github.com/tc39/proposal-regexp-legacy-features#updatelegacyregexpstaticproperties--c-s-startindex-endindex-capturedvalues-
+void RegExpConstructor::update_legacy_properties(Utf16String const& S, size_t startIndex, size_t endIndex, Vector<Utf16View>& capturedValues)
```

Comment:
Please use snek_case for variables/parameters.

- start_index
- end_index
- captured_values